### PR TITLE
Opt-in RBF must be strictly enabled or disabled before GBT can be called

### PR DESCRIPTION
**Pull request by wangchun**
Pool operators must strictly specify their preference to opt-in RBF.

### CODE REVIEW

**Comment by luke-jr** [comment]
Concept ACK, but see #3229 for past discussion.

**Comment by dcousens** [comment]
Concept ACK

**Comment by jtoomim** [comment]
This probably breaks unit tests. Perhaps you should add a check for regtest or testnet mode first? #3229 had the same issue.

**Comment by wangchun** [comment]
Updated with check for testnet and regtest

**Comment by sandakersmann** [comment]
ACK

**Comment by laanwj** [comment]
It would be better to update the RPC tests to pass the option explicitly instead of adding a flag to  `CChainParams`. This makes sure that the switching code is (or at least, can be) actually tested.

**Comment by sandakersmann** [comment]
This should be in 0.12

**Comment by luke-jr** [comment]
Agree with @laanwj on passing the flag in tests.

I also think it would be better to just reopen #3229 if the NACKs there are determined to be outdated/overruled...

@sandakersmann Way too late for any changes to 0.12 now.

**Comment by sandakersmann** [comment]
@luke-jr Then we should set default for -mempoolreplacement to false in 0.12.

**Comment by petertodd** [comment]
Weak NACK

I'm dubious about making it more difficult for users to mine out-of-the-box. p2pool is a thing that we want to encourage, and every additional setting that you must set to use it is a major barrier to adoption; there are literally dozens of settings that you might want to set.

A better way to deal with this kind of choice is choice between different Bitcoin protocol implementations with different philosophies behind them, e.g. Bitcoin Core and Bitcoin Classic.

Also, for this setting in particular, there was very little controversy associated with it pre-merging, including ACK's by Jeff Garzik and others with strong pro-zeroconf feelings. This is notably unlike things like OP_RETURN, where even among the technical community there was controversy pre-merging. The controversy post-merge seems to be based mainly on misunderstandings of how well zeroconf works; note my blog post on the subject: https://petertodd.org/2016/are-wallets-ready-for-rbf

**Comment by sandakersmann** [comment]
@petertodd 0-conf works well enough today. To bad you have to destroy this use case and make bitcoin less useful as a currency.

**Comment by gmaxwell** [comment]
@sandakersmann Please treat contributors here with respect. None of the contributors here want to make Bitcoin less useful. It sounds like you may be confused about the functionality in Bitcoin Core, please check out the FAQ: https://bitcoincore.org/en/faq/optin_rbf/

**Comment by sandakersmann** [comment]
@gmaxwell I have already checked out the FAQ and I'm not confused. When it comes to showing respect I'm sure I'm not the worst around here.

**Comment by luke-jr** [comment]
@sandakersmann If you are not confused, then you should be aware that RBF does not break unconfirmed/off-chain transactions at all. They continue to work fine with the same lack of security they already have. Anyhow, this is not the place to discuss the merits of specific policies.

**Comment by Aquentus** [comment]
utACK. Developers, with no modern mining or economic experience, should not be making policy choices in regards to defaults anyway, but only offer options so that the miners can choose policy aspects in a decentralized manner.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -242,6 +244,7 @@ class CRegTestParams : public CChainParams {
         vSeeds.clear();  //! Regtest mode doesn't have any DNS seeds.
 
         fMiningRequiresPeers = false;
+        fMiningRequiresConfiguration = false;
```
Comment (MarcoFalke) [inline src/chainparams.cpp]: Please remove those. Each difference between regnet and mainnet makes our tests a bit less useful.

You can adjust this in `qa/rpc-tests/test_framework/util.py` or even better in the respective `qa/rpc-tests` that cover  this.

**Comment by laanwj** [comment]
Closing this, there is no agreement to do this and furthermore, comments on the code are not being addressed.

**Pull request closed without merging**